src/main/java/picard/nio/PicardBucketUtils.java

```diff
@@ -110,10 +89,10 @@ public static PicardHtsPath getLocalTempFilePath(final String prefix, final Stri
      * @param relativePath The relative location for the new "directory" under the harcoded staging bucket with a TTL set e.g. "test/RevertSam/".
      * @return A PicardHtsPath object to a randomly generated "directory" e.g. "gs://hellbender-test-logs/staging/picard/test/RevertSam/{randomly-generated-string}/"
      */
-    public static PicardHtsPath getRandomGCSDirectory(final String relativePath){
+    public static IOPath getRandomGCSDirectory(final String relativePath){
         ValidationUtils.validateArg(relativePath.endsWith("/"), "relativePath must end in backslash '/': " + relativePath);
 
-        return PicardHtsPath.fromPath(PicardBucketUtils.randomRemotePath(GCLOUD_PICARD_STAGING_DIRECTORY + relativePath, "", "/"));
+        return PicardBucketUtils.randomRemotePath(PicardHtsPath.resolve(GCLOUD_PICARD_STAGING_DIRECTORY, relativePath), "", "/");
```

Comment:
Every direct reference to `GCLOUD_PICARD_STAGING_DIRECTORY` like this is bypassing the `GCloudUtils` methods such as `getTestStaging` that check the environment to determine if the user has specified an override staging area to use.